plugins/org.eclipse.sirius.tests.swtbot.support/src/org/eclipse/sirius/tests/swtbot/support/api/editor/SWTBotSiriusGefContextMenu.java

```diff
@@ -0,0 +1,137 @@
+import org.eclipse.swtbot.swt.finder.utils.MessageFormat;
+import org.eclipse.swtbot.swt.finder.utils.internal.Assert;
+import org.hamcrest.Matcher;
+
+/**
+ * An override of {@link SWTBotGefContextMenu} to have a fix directly in Sirius by waiting a fix in SwtBot.
+ * 
+ * @author Laurent Redor
+ *
+ */
+public class SWTBotSiriusGefContextMenu extends SWTBotGefContextMenu {
+    private final String text;
+
+    private final Control control;
+
+    /**
+     * Default constructor.
+     * 
+     * @param control
+     * @param text
+     */
+    public SWTBotSiriusGefContextMenu(final Control control, final String text) {
+        super(control, text);
+        this.text = text;
+        this.control = control;
+    }
+
+    // CHECKSTYLE:OFF Code copied from SWTBotGefContextMenu only with a change around "bar.notifyListeners(SWT.Hide, new
+    // Event());" in "invokeMenuInternal" to protect it.
+
+    @Override
+    public SWTBotGefContextMenu click() {
+            final boolean[] clicked = new boolean[1];
+            // must async exec in case the call opens a dialog
+            UIThreadRunnable.asyncExec(this.display, new VoidResult() {
+                @Override
+                public void run() {
+                    Menu menu = control.getMenu();
+                    invokeMenuInternal(menu, WidgetMatcherFactory.withMnemonic(text), clicked, true);
+                }
+            });
+            // use another sync exec to ensure that previous call was processed fully
+            UIThreadRunnable.syncExec(this.display, new VoidResult() {
+                @Override
+                public void run() {
+                    // do nothing, just wait for all events to be processed
+                }
+            });
+            if (!clicked[0]) {
+                throw new WidgetNotFoundException(text);
+            }
+            return this;
+    }
+
+    private boolean invokeMenuInternal(final Menu bar, final Matcher<? extends Widget> matcher, final boolean[] clickInitiated, final boolean recursive) {
+        if (bar != null) {
+            bar.notifyListeners(SWT.Show, new Event());
+            try {
+                MenuItem[] items = bar.getItems();
+                for (int i = 0; i < items.length; i++) {
+                    MenuItem menuItem = items[i];
+                    if (isSeparator(menuItem)) {
+                        continue;
+                    }
+                    if (matcher.matches(menuItem)) {
+                        clickInitiated[0] = true;
+                        click(menuItem);
+                        return true;
+                    }
+                    if (recursive) {
+                        if (invokeMenuInternal(menuItem.getMenu(), matcher, clickInitiated, recursive)) {
+                            return true;
+                        }
+                    }
+                }
+            } finally {
+                if (!bar.isDisposed()) {
```

Comment:
add a comment that the only change of this whole class is here

Reply:
Already done on comment line 53

Reply:
I will add another comment...